torch/utils/data/datapipes/iter/combining.py

```diff
@@ -181,7 +182,8 @@ class DemultiplexerIterDataPipe(IterDataPipe):
             classifier_fn: a function that maps values to an integer within the range [0, num_instances - 1] or None
             drop_none: defaults to False, if True, the function will skip over elements classified as None
             buffer_size: this defines the maximum number of inputs that the buffer can hold across all child
-                DataPipes while waiting for their values to be yielded
+                DataPipes while waiting for their values to be yielded.
+                Use -1 for the unlimited buffer
     """
     def __new__(cls, datapipe: IterDataPipe, num_instances: int,
                 classifier_fn: Callable[[T_co], int], drop_none: bool = False, buffer_size: int = 1000):
```

Comment:
Should we make it unlimited by default? Or a larger default? 

The question would be - how many users would run into memory issues if we make the default larger or unlimited.

Reply:
Let me add a global default buffer size (maybe 100_000) and let all buffers using it. If all buffer unlimited by default, the chance for OOM would be great. So,I would do a larger default for now. 

If more user requests are found, we are able to convert it to unlimited more smoothly (without breaking users' code)

Reply:
That's a good idea. Perhaps we can have a function that allows the users to set the global default too?

Reply:
Sure, then let's do another PR to do the both things:
- Add global default buffer size
- Add a way to set global default